### PR TITLE
fix: [Schema Evolution] Support multiple version tables and set basel…

### PR DESCRIPTION
**Pull request by zhixuanjia**
…ine version

Changes verified using GMS local deployment.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
+      if (!configProp.containsKey(EVOLUTION_SCRIPTS_LOCATION) || !configProp.containsKey(VERSION_TABLE)) {
+        throw new RuntimeException("script_directory and version_table are required entries in evolution.conf");
+      }
+
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load db evolution config");
+    }
+
+    //This is especially useful for production environments where running clean can be quite a career limiting move.
+    boolean disableClean = !configProp.containsKey(DISABLE_CLEAN) || Boolean.parseBoolean(configProp.getProperty(DISABLE_CLEAN));
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java:31]: Why don't we just always set this disableClean to true? Any use case for clean?

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
+    InputStream configFile = getClass().getClassLoader().getResourceAsStream(CONFIG_FILE);
+    Properties configProp = new Properties();
+
+    try {
+      configProp.load(configFile);
+
+      if (!configProp.containsKey(EVOLUTION_SCRIPTS_LOCATION) || !configProp.containsKey(VERSION_TABLE)) {
+        throw new RuntimeException("script_directory and version_table are required entries in evolution.conf");
+      }
+
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java:25]: very nit: unnecessary line

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java [hunk 3]
```diff
@@ -1,16 +1,42 @@
...
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load db evolution config");
+    }
+
+    //This is especially useful for production environments where running clean can be quite a career limiting move.
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java:30]: lol

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManagerTest.java [hunk 4]
```diff
@@ -46,6 +46,10 @@ public void testSchemaUpToDate() throws IOException {
 
     // V2__create_bar_entity_table.sql create metadata_entity_bar table.
     assertTrue(checkTableExists("metadata_entity_bar"));
+
+    // Make sure version table is created.
+    assertTrue(checkTableExists("my_version_table"));
+    _server.createSqlUpdate("DROP TABLE  my_version_table").execute();
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManagerTest.java:52]: very nit: extra space between TABLE and my_version_table

**Review by jsdonn — CHANGES_REQUESTED** [review]

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java (hunk 1 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java:31]: Clean is useful when we want to have a fresh db to start with, specially when running unit tests. We'll keep this as internal config and not mention it on the onboarding doc.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java (hunk 2 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java]: deleted.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java (hunk 1 above)
Comment (jsdonn) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/FlywaySchemaEvolutionManager.java:30]: Sounds good to me.

**Review by jsdonn — APPROVED** [review]
LGTM. will review onboarding steps later.

**Pull request merged**